src/Infrastructure.EntityFramework/Repositories/CollectionRepository.cs

```diff
@@ -180,6 +180,110 @@ where cg.CollectionId.Equals(id)
         }
     }
 
+    public async Task<ICollection<Tuple<Core.Entities.Collection, CollectionAccessDetails>>> GetManyByOrganizationIdWithAccessAsync(Guid organizationId)
+    {
+        var collections = await GetManyByOrganizationIdAsync(organizationId);
+        using (var scope = ServiceScopeFactory.CreateScope())
+        {
+            var dbContext = GetDatabaseContext(scope);
+            var groups =
+                from cg in dbContext.CollectionGroups
+                where cg.Collection.OrganizationId == organizationId
+                group cg by cg.CollectionId into g
```

Comment:
Did these `group .. by` statement work for you in EF? I was recently trying this myself in the members work and EF core was complaining about it not being translatable to sql. I was forced to do it in after awating: https://github.com/bitwarden/server/blob/d24f0bbfbc1ee1fbc216d1be7231cfbf796bef28/src/Infrastructure.EntityFramework/Repositories/OrganizationUserRepository.cs#L325-L328

Reply:
That's interesting, did you use the  same `group .. by .. into` syntax? I haven't run into any issues with mine, as long as I selected the new `IGrouping` at the end.

Reply:
Hmm, I'm pretty sure I used the `.. into` syntax, but possibly not or I had made some other mistake. But, if this is working as expected for you, it was likely just a problem I had so this PR should be good to go